klayout/drc/rule_decks/hres.drc

```diff
@@ -0,0 +1,150 @@
+# frozen_string_literal: true
+
+################################################################################################
+# Copyright 2022 GlobalFoundries PDK Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################################
+
+if FEOL
+  #================================================
+  #----------------H POLY RESISTOR-----------------
+  #================================================
+
+  logger.info('Starting HRES derivations')
+  hres_poly = poly2.interacting(pplus).interacting(sab).interacting(res_mk).interacting(resistor)
+  hres1_poly = poly2.interacting(pplus).interacting(sab).interacting(res_mk)
+
+  # Rule HRES.1: Minimum space. Note : Merge if the spacing is less than 0.4 um. is 0.4µm
+  logger.info('Executing rule HRES.1')
+  hres1_l1 = resistor.interacting(hres1_poly).space(0.4.um, euclidian)
+  hres1_l1.output('HRES.1', 'HRES.1 : Minimum space. Note : Merge if the spacing is less than 0.4 um. : 0.4µm')
```

Comment:
not sure why we need to repeat the constraint twice?

Reply:
We have 2 rules: 

- `HRES.1`: We need to check spacing between `resistor` polygons that construct HRES. So We have defined it as `resistor.interacting(hres1_poly)` which `hres1_poly` is a poly2 inside HRES.

- `HRES.3`: We need to check spacing between `poly2` polygons but we need to make sure that this poly2 is a HRES poly so we have defined `hres_poly` which checks if it's interacting with resistor marker or not.

Reply:
I have optimized it to be clear.

Reply:
Thanks @FaragElsayed2